include/highfive/bits/H5Dataspace_misc.hpp

```diff
@@ -150,12 +150,14 @@ inline DataSpace DataSpace::From(const ScalarValue& scalar) {
 
 template <typename Value>
 inline DataSpace DataSpace::From(const std::vector<Value>& container) {
-    return DataSpace(details::get_dim_vector<Value>(container));
+    auto dims = details::inspector<std::vector<Value>>::getDimensions(container);
+    return DataSpace(std::vector<size_t>(dims.begin(), dims.end()));
```

Comment:
Adding a `std::array` ctor to DataSpace makes sense?

Reply:
yes ;)